es6-shim.js

```diff
@@ -756,6 +771,64 @@
   });
   addIterator(ArrayIterator.prototype);
 
+  var ObjectIterator = function (object, kind) {
+    this.object = object;
+    this.i = 0;
+    // Don't generate keys yet.
+    this.array = null;
+    this.kind = kind;
+  };
+
+  defineProperties(ObjectIterator.prototype, {
+    next: function () {
+      var key, array = this.array;
+      if (!(this instanceof ObjectIterator)) {
+        throw new TypeError('Not an ObjectIterator');
+      }
+
+      if (typeof array !== 'undefined') {
+        // Keys not generated
```

Comment:
Could we put the "key generation" logic in a separate, closured function? Something like `this.array = getAllKeys(this.object);`